### DISCUSSION TITLE
Advice on a change to fromZigbee.js

### DISCUSSION THREAD

**Discussion by bayview252** [Koenkk/zigbee2mqtt#7991] [Devices]
Hi, thanks to the developers and contributors for this great ecosystem!  I'm just after some guidance on pushing a change to zigbee-herdsman-converters/converters/fromZigbee.js, and whether the change should be made to a command or whether to provide it as a device-specific command? So do I push Option 1 or 2?

**What I'm trying to solve:** I've recently bought som RGBGenie 5028 RGBW 4 Zone Remotes, and have noticed that the converter (RGBgenie ZB-5028) in /devices/rgbgenie.js is not passing the full message payload received when the hue_move button is pressed through to Home Assistant.  Should be receiving a payload of "movemode": 0 or 1, and rate: 0 or (1 - 63) which should be translated as movemode: 0 "hue_move_stop_(1-4)" or "hue_move_(1-4)" with a rate: 1-64 in the attributes. The issue is with command_move_hue in fromZigbee.js which doesn't interpret and pass the attributes. All that is received is action: hue_move_1/2/3/4. The Z2M debug log showed that the msg received included the movemode and rate attributes. I've tested my proposed change and it works as I expected.

**Why I'm asking**
As I'm not a software engineer, I'm worried that I will break working remote integrations for other users that expect to just receive a hue_move_1/2/3/4 action by introducing a new hue_stop_1/2/3/4 "action".

 **Current fromZigbee.command_move_hue:**
```
    command_move_hue: {
        cluster: 'lightingColorCtrl',
        type: 'commandMoveHue',
        convert: (model, msg, publish, options, meta) => {
            const payload = {action: postfixWithEndpointName('hue_move', msg, model)};
            addActionGroup(payload, msg, model);
            return payload;
        },
    },
```

**Option 1 Proposed change: - modify the command_move_hue command.**
```
command_move_hue: {
    cluster: 'lightingColorCtrl',
    type: 'commandMoveHue',
    convert: (model, msg, publish, options, meta) => {
      const move_stop = msg.data.movemode == 1 ? 'move' : 'stop';
      const action = postfixWithEndpointName(`hue_${move_stop}`, msg, model); 
      const payload = {action, action_rate: msg.data.rate};
        addActionGroup(payload, msg, model);
        return payload;
    },
  },
```

**Option 2 Proposed Change: - create a device specfiic ZB5028_command_move_hue command**
Same as above, but appended to fromZigbee.js as ZB5028_command_move_hue and change the fromZigbee command in  the ZB5028 device definition within devices/rgbgenie.js from "fz.command_move_hue" to "fz.ZB5028_command_move_hue"

Thanks!

**Comment by Koenkk** [comment]
The preferred way is: `const payload = {action, action_rate: msg.data.rate};` (same as https://github.com/Koenkk/zigbee-herdsman-converters/blob/c7575b04d4a7e8df3079018b16d8d44a14facc01/converters/fromZigbee.js#L1065)

**Reply by bayview252** [reply]
Thanks, yes, I copied the wrong block, that original payload line threw an error, and I fixed it with vi in the docker container, but didn't copy it back to the file I wrote it in. I've changed in my initial post now.
But the question still remains, which option should I push? 
Change the main move_hue_command? OR Create a device-specific change? 
Thanks for the reply.

**Reply by Koenkk** [reply]
Modify the main move_hue_command.

**Reply by bayview252** [reply]
will do.
